ecl/hqlcpp/hqlckey.cpp

```diff
@@ -1595,6 +1595,9 @@ ABoundActivity * HqlCppTranslator::doBuildActivityKeyedDistribute(BuildCtx & ctx
 
 ABoundActivity * HqlCppTranslator::doBuildActivityKeyDiff(BuildCtx & ctx, IHqlExpression * expr, bool isRoot)
 {
+    if (expr->hasAttribute(_disallowed_Atom))
+        throwError(HQLERR_DatafileRequiresSigned);
```

Comment:
This should be calling reportError instead - to provide more context about where the error occurred in the original source.
(One reason why commoning code into a function is a good idea)